src/SFML/System/Android/Activity.hpp

```diff
@@ -91,7 +91,9 @@ struct ActivityStates
     LogcatStream logcat;
 };
 
-SFML_SYSTEM_API ActivityStates* getActivity(ActivityStates* initializedStates=NULL, bool reset=false);
+SFML_SYSTEM_API ActivityStates*& getActivityStatesPtr();
+SFML_SYSTEM_API void resetActivity(ActivityStates* initializedStates);
+SFML_SYSTEM_API ActivityStates& getActivity();
```

Comment:
This is the most significant change, I split a single monolithic function into two smaller ones with a more well-defined and specific responsibility. `getActivityStatesPtr` could be hidden in the `.cpp` but I wasn't sure whether providing `static` linkage for the function itself would screw up things.